src/common/SPELLS/index.ts

```diff
@@ -77,16 +77,18 @@ const ABILITIES = {
   ...SHADOWLANDS,
 } as const;
 
+type SpellCollection = SpellList & {
+  maybeGet: (key: string | number | undefined) => Spell | undefined;
+};
 // If you remove this indexById you can see what spells are undefined.
 // But you'll get a lot of other errors.
 // We should type indexById properly some day to make this standard.
 // And then fix all those errors.
 // Which will prevent bugs.
-const InternalSpellTable = indexById(ABILITIES);
+const InternalSpellTable = indexById(ABILITIES) as SpellCollection;
 // assignment is used here to avoid potential performance pitfalls when
 // compiling the spread operator on large objects.
-InternalSpellTable.maybeGet = (key: string | number | undefined): Spell | undefined =>
-  key ? InternalSpellTable[key] : undefined;
+InternalSpellTable.maybeGet = (key) => (key ? InternalSpellTable[key] : undefined);
```

Comment:
Could this be defined as a separate export and the current uses of `SPELLS.maybeGet` refactored to use the separate export instead? This was very much relying on the fact that `indexById` was previously returning `any/unknown`.